src/api/index.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 RoadieHQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef, DiscoveryApi } from '@backstage/core';
+
+export const buildKiteApiRef = createApiRef<BuildKiteApi>({
+  id: 'plugin.buildkite.service',
+  description: 'Used by the BuildKite plugin to make requests',
+});
+
+const DEFAULT_PROXY_PATH = '/buildkite/api';
```

Comment:
Do we need the /api at the end ? I would have thought that the /api prefix for the proxy itself makes that clear enough i.e. the whole path would be `/api/proxy/buildkite` ? and that would forward to api.buildkite.com ? (the `v2` part maybe could be a const in the BuildKiteApi if it's the only option.)

Reply:
I would leave it that way, it is the same in CircleCI and Jenkins plugins. Also it is just a fallback for proxyPath taken from discoveryApi where all plugins follow this pattern .